### PR TITLE
#9322: sandbox injection changes

### PR DESCRIPTION
**Pull request by fungairino**
## What does this PR do?

- Part of #9322
- Ensures iframe removed interval is cleared
- Sets fetchPriority of sandbox script to high to attempt to make loading the iframe faster / higher priority for the browser when resource constrained.

## Future Work

- _Work for the issue/ticket that will be in a follow-up PR_

## Checklist

- [N/A ] Added jest or playwright tests and/or storybook stories

For more information on our expectations for the PR process, see the
[code review principles doc](https://www.notion.so/pixiebrix/Code-Review-Principles-1ce7276b82a84d2a995d55ad85e1310d?pvs=4)

### CODE REVIEW

File: src/utils/injectIframe.tsx [hunk 1]
```diff
@@ -75,10 +60,17 @@ async function _injectIframe(
...
+  const removedPoll = setInterval(() => {
+    if (!document.documentElement.contains(shadowElement)) {
+      clearInterval(removedPoll);
+      resolve("removed");
+    }
+  }, 100);
+
+  const result = await iframeLoadPromise;
+
+  clearInterval(removedPoll);
```
Comment (fungairino) [inline src/utils/injectIframe.tsx:73]: This ensures we clearInterval on the elementRemoved interval, which we were previously not doing

File: src/sandbox/sandbox.html [hunk 2]
```diff
@@ -16,4 +16,4 @@
   -->
 <!doctype html>
 <meta charset="utf-8" />
-<script src="sandbox.js"></script>
+<script src="sandbox.js" fetchpriority="high"></script>
```
Comment (fungairino) [inline src/sandbox/sandbox.html:19]: The idea is that hopefully the browser will prioritize loading the sandbox script over other resources in the browser when dealing with a resource constrained environment.

Comment (fungairino) [reply src/sandbox/sandbox.html:19]: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#fetchpriority

File: src/utils/injectIframe.tsx (hunk 1 above)
Comment (mnholtz) [reply src/utils/injectIframe.tsx:73]: Did you mean to remove the Promise.race here? It looks like `result` will never be `"removed"` in any case.

File: src/sandbox/sandbox.html (hunk 2 above)
Comment (mnholtz) [reply src/sandbox/sandbox.html:19]: Let's make sure to follow-up with checking data dog to see if there is any impact on frequency of sandbox messaging errors.

File: src/utils/injectIframe.tsx (hunk 1 above)
Comment (fungairino) [reply src/utils/injectIframe.tsx:73]: Yes. Instead of using Promise.race, we resolve the same promise

**Comment by fungairino** [comment]
@fregante do you have context on why we inject a new sandbox for each frame on the page rather than just communicating with the top-level iframe sandbox? 

We are looking into timeout errors when communicating with the sandbox and where we could possibly improve load performance for users with resource constrained devices

**Comment by fregante** [comment]
The sandbox doesn't have the extension's messaging API so you can only reach it via postMessage; can't do that across cross-origin frames. I suppose you could attempt to use `parent.document.querySelector('.pb-sandbox')` and default to the current behavior when that fails

**Review by grahamlangford — APPROVED** [review]

File: src/utils/injectIframe.tsx (hunk 1 above)
Comment (mnholtz) [reply src/utils/injectIframe.tsx:73]: Ah, silly mistake. I see it now

**Pull request merged**